sdk/dotnet/Pulumi/Deployment/Deployment.cs

```diff
@@ -93,27 +89,32 @@ private Deployment()
             _stackName = stack;
             _projectName = project;
 
-            _options = new Options(
-                queryMode: queryModeValue, parallel: parallelValue, pwd: pwd,
-                monitor: monitor, engine: engine, tracing: tracing);
+            Serilog.Log.Debug("Creating Deployment Engine and Monitor.");
+            this.RpcDispatcher = new GrpcDispatcher(engine, monitor);
+            Serilog.Log.Debug("Created Deployment Engine and Monitor.");
 
-            Serilog.Log.Debug("Creating Deployment Engine.");
-            this.Engine = new Engine.EngineClient(new Channel(engine, ChannelCredentials.Insecure));
-            Serilog.Log.Debug("Created Deployment Engine.");
-
-            Serilog.Log.Debug("Creating Deployment Monitor.");
-            this.Monitor = new ResourceMonitor.ResourceMonitorClient(new Channel(monitor, ChannelCredentials.Insecure));
-            Serilog.Log.Debug("Created Deployment Monitor.");
+            _runner = new Runner(this);
+            _logger = new Logger(this, this.RpcDispatcher);
+        }
 
+        /// <summary>
+        /// This constructor is called from <see cref="TestAsync{TStack}"/> with
+        /// a mocked monitor and dummy values for project and stack.
+        /// </summary>
+        private Deployment(IRpcDispatcher rpcDispatcher)
+        {
+            _isDryRun = true;
```

Comment:
Why is isDryRun set to true here?

Reply:
A unit test is like a preview, isn't it?

Reply:
Not necessarily: at least in other SDKs, the behavior of `apply` and some of the marshalling code differs in a preview. I think we want this to be configurable, as it is in other SDKs.

Reply:
@pgavlin Good point! I introduced options for preview, project, and stack akin to other SDKs.